core/base/kdTree/KDTree.h

```diff
@@ -64,14 +64,18 @@ namespace ttk {
                     const int &ptNumber,
                     const int &dimension,
                     const std::vector<std::vector<dataType>> &weights = {},
-                    const int weight_number = 1);
+                    const int weight_number = 1,
```

Comment:
I'd recommend to take the habit to use references for function parameters (unless pointers are necessary), even for integers.
`const int weight_number = 1` -> `const int &weightNumber = 1`
(also, we usually don't use underscores for names in TTK, but a camel wording)